tools/walletextension/api/routes.go

```diff
@@ -216,3 +241,124 @@ func submitViewingKeyRequestHandler(walletExt *walletextension.WalletExtension,
 		return
 	}
 }
+
+func authenticateRequestHandler(walletExt *walletextension.WalletExtension, userConn userconn.UserConn) {
+	// check if the text is well-formed and extract userID and address
+	body, err := userConn.ReadRequest()
+	if err != nil {
+		return
+	}
+
+	var reqJSONMap map[string]string
```

Comment:
this needs some comments.
I don't quite follow what's going on here.

Based on the design there should be 2 fields in the json:
- The text that was signed
- the signature

What I imagined is that:
1. parse the text and extract the userId and the account.
2. from the db, fetch the VK entry for the userId
3. Check the signature of the account over text
4. Register the account against the VK